packages/components/src/select-control/stories/index.tsx

```diff
@@ -32,20 +31,33 @@ const meta: ComponentMeta< typeof SelectControl > = {
 };
 export default meta;
 
-const SelectControlWithState: ComponentStory< typeof SelectControl > = ( {
-	onChange,
-	...args
-} ) => {
-	const [ selection, setSelection ] =
-		useState< ComponentProps< typeof SelectControl >[ 'value' ] >();
+const SelectControlWithState: ComponentStory< typeof SelectControl > = (
+	props
+) => {
+	const [ selection, setSelection ] = useState< string[] >();
+
+	if ( props.multiple ) {
```

Comment:
This felt slightly hacky, but I couldn't figure out a much more elegant way.

Basically, since the `value` of `SelectControl` is controlled with the `selection` variable from internal state, I couldn't find an easy way to get TypeScript to automatically understand when `selection` can be used as a `string` and when it can be used as a `string[]`.

Therefore, I decided to have `selection` to always be `string[]`, and I've added some code to "adapt" the array to a single value for when `multiple !== true`.

In doing do, I also had to "override" the `multiple` prop passed to `SelectControl` to avoid a TypeScript error.

Reply:
For stories this seems reasonable.

Reply:
Agreed with Jon here, I believe it also makes it quite explicit and transparent what we're aiming to do, which is always useful for a story/test/example.